estimator_checks test for 1 sample and 1 feature

sotlampr opened the pull request:
Issue [#5380](https://github.com/scikit-learn/scikit-learn/issues/5380)

sotlampr committed:
estimator_checks test for 1 sample and 1 feature

amueller commented on sklearn/utils/estimator_checks.py:
```diff
@@ -411,13 +411,14 @@ def check_fit2d_1sample(name, Estimator):
+    except:
+        # The message should say something about only one sample
+        msg = "sample"
+        assert_raise_message(ValueError, msg, estimator, X, y)
```
that is not right, sorry I was not having a proper look. We already ran ``fit`` and caught the exception. You need to check the exception similar to this: https://github.com/sotlampr/scikit-learn/blob/master/sklearn/utils/estimator_checks.py#L326

sotlampr committed:
Catch exceptions as in ''check_estimator_sparse_data''

amueller commented:
Why isn't travis running on this PR... weird...

amueller commented:
Well the errors are of two kinds: one is that algorithm throw "the number of classes in y must be greater than one" which I think is a helpful error message and we should allow. The other one is by algorithms that don't throw a helpful error message. This should either require two samples (using the ``min_sample`` option in``check_X_y`` or ``check_arrays``) or should be made to work with one sample.

sotlampr committed:
fixed error message in SGDClassifier, broadened check_fit2d_1sample message check

sotlampr committed:
changed various estimators to comply with tests

sotlampr committed:
Fixed typos

sotlampr committed:
fixed most of the test errors

sotlampr committed:
fixed redundancy in GaussianProcess

sotlampr committed:
modified ''ensure_min_samples'' in spectral clustering

sotlampr commented:
I fixed most of the errors by setting ''ensure_min_samples'' or slightly changing the error message on the failing estimators except:
* in  the ''cluster.mean_shift_.py'' when ''bandwidth'' is set to None the ''estimate_bandwidth'' function fails  if the samples are too few. As this number is somewhat arbitrary, I added a check for this that raises an appropriate error.

There are now 2 errors in the test:
* In RANSAC the message is OK but it is not useful, I don't see how I can change that.
* There is something wrong with gaussian process, I guess it was there before.

sotlampr commented:
I will make a new clean attempt at this.
